riptide-core/src/main/java/org/zalando/riptide/AbstractCancelableCompletableFuture.java

```diff
@@ -0,0 +1,22 @@
+package org.zalando.riptide;
+
+import java.util.concurrent.CompletableFuture;
+
+import static org.zalando.riptide.CancelableCompletableFuture.preserveCancelability;
+
+public abstract class AbstractCancelableCompletableFuture<T> extends CompletableFuture<T> {
```

Comment:
I this this is one level more abstract than necessary. Especially, its only method strongly couples it to its default implementation.

Reply:
Oh, I see it is because of Java 9.